cmd/cosign/cli/copy.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"github.com/sigstore/cosign/pkg/cosign"
+)
+
+func Copy() *ffcli.Command {
+	var (
+		flagset     = flag.NewFlagSet("cosign copy", flag.ExitOnError)
+		srcImgFlag  = flagset.String("source", "", "the image (and associated signatures) to copy")
+		dstImgFlag  = flagset.String("destination", "", "the name of the copied image")
+		sigOnlyFlag = flagset.Bool("sig-only", false, "only copy the image signature")
+		forceFlag   = flagset.Bool("force", false, "overwrite destination image(s), if necessary")
+	)
+	return &ffcli.Command{
+		Name:       "copy",
+		ShortUsage: "cosign copy -source <source image> -destination <destination image>",
+		ShortHelp:  `Copy the supplied container image and signatures.`,
+		LongHelp: `Copy the supplied container image and signatures.
+
+EXAMPLES
+  # copy a container image and signature
+  cosign copy -source example.com/foo:latest -destination example.com/bar:oldest
```

Comment:
This might not be the best example since it's the same repo. It would "work" but isn't really any different than a crane cp, since the signature doesn't actually move. Maybe two repos?

Reply:
done